ui-tests/src/test/java/org/openmrs/reference/page/AttachmentsPage.java

```diff
@@ -0,0 +1,58 @@
+package org.openmrs.reference.page;
+
+import org.openmrs.uitestframework.page.Page;
+import org.openqa.selenium.By;
+
+public class AttachmentsPage extends Page {
+	
+	private static final By ATTACH_FILE = By.cssSelector("#visit-documents-dropzone");
+	private static final By ADD_CAPTION = By.cssSelector("textarea.ng-pristine");
+	private static final By UPLOAD_FILE = By.cssSelector("button.confirm:nth-child(1)");
+	private static final By CLEAR_FORMS = By.cssSelector("button.ng-binding:nth-child(2)");
+	public static final By CAMERA_ICON = By.cssSelector(".left > i:nth-child(1)");
+	private static final By CAMERA_BUTTON = By.cssSelector("a.button:nth-child(1) > i:nth-child(1)");
+	public static final By SAVE_IMAGE = By.cssSelector("a.confirm:nth-child(3)");
```

Comment:
have you come across `element[attribute='attribute-value']` such as `By.cssSelector("button[ng-click='clearForms()']");` for `<button ng-click="clearForms()" class="ng-binding">Clear forms</button>`

I would rather prefer using those stable ways to getting to elements than using the element tree which can easily change.

Reply:
Surely had never but took a look into this and made the recommendation in the updated commit